pysat/tests/classes/cls_instrument_iteration.py

```diff
@@ -310,229 +312,226 @@ def test_set_bounds_with_frequency_and_width(self):
+    def test_iterate_bounds_with_frequency_and_width(self, starts, stops, step,
+                                                     width, by_date):
         """Test iterate via date with mixed step/width, excludes stop date.
 
         Parameters
         ----------
-        values : tuple of size 4
-            Defines the iteration bounds: (start, stop, step, width). start
-            and stop should be datetime, step and width should be int.
+        starts : dt.datetime or list of dt.datetime
+            The start date for iterations, or dates for iteration over multiple
+            segments.
+        stops : dt.datetime or list of dt.datetime
+            The start date for iterations, or dates for iteration over multiple
+            segments.
+        step : int
+            The step size for the iteration bounds.
+        width : int
+            The width of the iteration bounds.
         by_date : bool
             If True, iterate by date.  If False, iterate by filename.
 
         """
 
-        out = self.support_iter_evaluations(values, for_loop=True,
+        out = self.support_iter_evaluations(starts, stops, step, width,
+                                            for_loop=True,
                                             by_date=by_date)
         self.verify_iteration(out, reverse=False, inclusive=False)
 
         return
 
-    @pytest.mark.parametrize("values", [(dt.datetime(2009, 1, 1),
-                                         dt.datetime(2009, 1, 4),
-                                         2, 2),
-                                        (dt.datetime(2009, 1, 1),
-                                         dt.datetime(2009, 1, 4),
-                                         3, 1),
-                                        (dt.datetime(2009, 1, 1),
-                                         dt.datetime(2009, 1, 4),
-                                         1, 4),
-                                        (dt.datetime(2009, 1, 1),
-                                         dt.datetime(2009, 1, 5),
-                                         4, 1),
-                                        (dt.datetime(2009, 1, 1),
-                                         dt.datetime(2009, 1, 5),
-                                         2, 3),
-                                        (dt.datetime(2009, 1, 1),
-                                         dt.datetime(2009, 1, 5),
-                                         3, 2)])
+    @pytest.mark.parametrize(
+        "starts,stops,step,width",
+        [(dt.datetime(2009, 1, 1), dt.datetime(2009, 1, 4), 2, 2),
+         (dt.datetime(2009, 1, 1), dt.datetime(2009, 1, 4), 3, 1),
+         (dt.datetime(2009, 1, 1), dt.datetime(2009, 1, 4), 1, 4),
+         (dt.datetime(2009, 1, 1), dt.datetime(2009, 1, 5), 4, 1),
+         (dt.datetime(2009, 1, 1), dt.datetime(2009, 1, 5), 2, 3),
+         (dt.datetime(2009, 1, 1), dt.datetime(2009, 1, 5), 3, 2)])
```

Comment:
`starts` should be a variable in the function.  Is there a reason `stops` couldn't be a parametrize call above the step/width call?  This would change the inputs a little.